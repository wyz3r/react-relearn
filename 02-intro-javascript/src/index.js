
const useState = (valor) => {
    return [valor, ()=> {console.log('hello, world')}]
}

const [name, setName] = useState('Goku')
console.log(name)
setName()
