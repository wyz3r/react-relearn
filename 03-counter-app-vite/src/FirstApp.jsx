import React from 'react'; // Necesario si usas JSX

import PropTypes from 'prop-types'

 
export const FirstApp = ({title, subTitle, name}) => {


    
    
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      {/* <h1>{getResult(1 , 5)} </h1> */}
      {/* <code>{JSON.stringify(name)}</code> */}
      <p> {subTitle}</p>
      <p> {subTitle}</p>
      
      <p> {name}</p>


    </>
  );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    
}

FirstApp.defaultProps = {
    
    subTitle: 'esto es sub',
    name:'no hay name '
}
