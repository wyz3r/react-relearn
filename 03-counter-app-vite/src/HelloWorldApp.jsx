// export function App () {

//     return  (<h1>hello world !!!!! </h1>)

// }


export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  )
}
