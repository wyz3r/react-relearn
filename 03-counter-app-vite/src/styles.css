html,body {
    background: #21232a;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 13.rem;
    border-radius: 4px;
}