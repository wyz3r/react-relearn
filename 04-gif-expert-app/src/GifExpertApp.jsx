import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (newCategory) => {
    // setCategories( (cat ) => ['Valorant', ...cat, ])
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((c) => (
        <GifGrid key={c} category={c} />
      ))}
    </>
  );
};
