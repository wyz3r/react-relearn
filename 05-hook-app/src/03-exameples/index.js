export * from './PokemonCard'
export * from './LoadingMessage'