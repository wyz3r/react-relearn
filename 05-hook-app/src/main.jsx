import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
// import { HooksApp } from './HooksApp.jsx'
import './index.css'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import s{ Layout } from './05-useLayoutEffect.jsx/layout'
// import { Memorize } from './06-memo/Memorize'
// import { MemoHook } from './06-memo/MemoHook'
import { CallbackHook } from './06-memo/CallbackHook'
import { Padre } from './07-tarea-memo/Padre'
// import { CounterApp } from './01-useState/CounterApp.jsx'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook.jsx'
// import { SimpleForm } from './02-useEffect/SimpleForm.jsx'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook.jsx'
// import { MultipleCustomHooks } from './03-exameples/MultipleCustomHooks.jsx'


createRoot(document.getElementById('root')).render(
  <StrictMode>
    <Padre />
  </StrictMode>,
)
