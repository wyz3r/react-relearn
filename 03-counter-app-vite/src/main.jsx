import React from "react";
import ReactDOM from "react-dom/client";
// import { FirstApp } from "./FirstApp";

import './styles.css';
import { CounterApp } from "./CounterApp";

// import  {HelloWorldApp} from  "./HelloWorldApp"

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp  /> */}
    <CounterApp value={3}/>
  </React.StrictMode>
);
