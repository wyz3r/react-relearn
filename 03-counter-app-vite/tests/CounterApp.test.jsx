import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("<CounterApp/>  component", () => {
    const initialValue = 10
  test("should match with the snapshot", () => {
   const {container} =  render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot()
  });

  test("should validate the first value start = 100", () => {
    
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy()
    expect(screen.getByRole('heading',{level:2}).innerHTML).toContain('100')
   });

   test('should increment the counter value when click button +1', () => { 
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('11')).toBeTruthy()
    })

    test('should decrement the counter value when click button -1', () => { 
      render(<CounterApp value={initialValue} />);
      fireEvent.click(screen.getByText('-1'))
      expect(screen.getByText('9')).toBeTruthy()
      })

    test('Reset button should work', () => { 

      render(<CounterApp value={initialValue} />);
      fireEvent.click(screen.getByText('+1'))
      fireEvent.click(screen.getByText('+1'))
      // fireEvent.click(screen.getByText('Reset'))
      
      fireEvent.click(screen.getByRole('button',{name:'btn-reset'}))
      expect(screen.getByText(initialValue).innerHTML).toBe(initialValue.toString())


     })
  
});
