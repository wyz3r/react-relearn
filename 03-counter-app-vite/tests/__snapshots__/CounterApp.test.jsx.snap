// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp/>  component should match with the snapshot 1`] = `
<div>
  <h1>
    Counter app
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
